ownership-chain/runtime/src/tests/version_tests.rs

```diff
@@ -0,0 +1,105 @@
+use crate::VERSION;
+use std::{
+	io::{Error, ErrorKind},
+	num::ParseIntError,
+};
+
+/// Converts a version string in the format "major.minor.patch" to an integer.
+/// Each part of the version (major, minor, patch) must be a number less than 100.
+/// The resulting integer is in the format major*10000 + minor*100 + patch.
+///
+/// # Arguments
+/// * `version` - A string slice representing the version.
+///
+/// # Returns
+/// A `Result` containing either the integer representation of the version
+/// or an `Error` if the input format is incorrect or any part of the version
+/// is 100 or greater.
+///
+/// # Examples
+/// ```
+/// let version = "1.2.3";
+/// let int_version = version_to_int(version).unwrap();
+/// assert_eq!(int_version, 10203);
+/// ```
+fn parse_semantic_version(version: &str) -> Result<u32, Error> {
```

Comment:
since this is used in the test, we don't really care about error handling, it's okay if it panics. it seems slightly overengineered